ppa/archive/management/commands/generate_textcorpus.py

```diff
@@ -191,10 +28,36 @@ def add_arguments(self, parser):
             "The default of -1 considers ALL documents.",
         )
 
+    def iter_solr(self, nsize=10, item_type='page'):
+        i=0
+        q=SolrQuerySet().search(item_type=item_type)
+        total = q.count()
+        for i in progressbar(range(0, total, nsize)):
+            q.set_limits(i,i+nsize)
+            yield from q
+
+    def iter_pages(self):
+        for d in self.iter_solr(item_type='page'):
+            yield {k:v for k,v in d.items() if k in self.PAGE_OUTPUT_FIELDS}
+
+    def iter_works(self):
+        for d in self.iter_solr(item_type='work'):
+            yield d
 
     def handle(self, *args, **options):
-        with logging_disabled():
-            SolrCorpus(
-                path=options["path"],
-                doc_limit=options["doc_limit"],
-            ).save()
\ No newline at end of file
+        path = options['path']
+        os.makedirs(path, exist_ok=True)
+        path_meta = os.path.join(path,'metadata.jsonl')
+        path_texts = os.path.join(path,'pages.jsonl')
+        
+        with jsonlines.open(path_meta,'w') as of_meta:
+            for i,d in enumerate(self.iter_works()):
+                of_meta.write(d)
+                if options['doc_limit']>0 and i+1>=options['doc_limit']: 
+                    break
```

Comment:
I missed that you were generating the metadata export in this script as well. Would like to align with the admin export (minus internal fields, but I guess that can be a later step.

Reply:
I don't think the metadata is large enough to merit jsonl and don't think it needs to be the same

Reply:
Is the solr and admin metadata (or another export command) guaranteed to be in sync?

Agreed, metadata can be json easily, will change back.